bin/disco_aws.py

```diff
@@ -351,10 +362,27 @@ def run():
     elif args.mode == "exec":
         instances = instances_from_args(aws, args)
         exit_code = 0
-        for instance in instances:
-            _code, _stdout = aws.remotecmd(instance, [args.command], user=args.user, nothrow=True)
-            sys.stdout.write(_stdout)
-            exit_code = _code if _code else exit_code
+        if args.concurrent and len(instances) > 1:
+            thread_objs = []
+            for instance in instances:
+                exec_obj = {}
+                thread = Thread(target=exec_in_thread,
```

Comment:
have you tested this? I have had strange behavior using threads in the past (long time ago) and have exclusively only used processes in python in recent years.

Reply:
yep, looks like ptyhon GIL is still in effect... aka only one thread is executed at any given time even if you spawn multiple threads, which should be fine in I/O bound tasks, but degrades performance in CPU bound tasks

Reply:
for this particular case it's doing this `aws.remotecmd()` thing which sounds more I/O bound, so 👍